pkg/scheduler/framework/plugins/helper/node_affinity.go

```diff
@@ -72,7 +72,9 @@ func PodMatchesNodeSelectorAndAffinityTerms(pod *v1.Pod, node *v1.Node) bool {
 // nodeMatchesNodeSelectorTerms checks if a node's labels satisfy a list of node selector terms,
 // terms are ORed, and an empty list of terms will match nothing.
 func nodeMatchesNodeSelectorTerms(node *v1.Node, nodeSelectorTerms []v1.NodeSelectorTerm) bool {
-	return v1helper.MatchNodeSelectorTerms(nodeSelectorTerms, node.Labels, fields.Set{
-		"metadata.name": node.Name,
-	})
+	matches, err := corev1.MatchNodeSelectorTerms(node, nodeSelectorTerms)
+	if err != nil {
+		klog.ErrorS(err, "Couldn't parse all NodeSelector terms", "node", klog.KObj(node), "terms", nodeSelectorTerms)
```

Comment:
this shouldn't be associated to the Node, as it wouldn't vary
The pod should be associated though.

Reply:
Good point, updated. Can you re-lgtm? Thanks!